extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSourceFactory.java

```diff
@@ -64,11 +73,17 @@ public DataSource createSource(DataFlowRequest request) {
 
         var source = request.getSourceDataAddress();
 
-        var secretToken = new AwsSecretToken(source.getProperty(ACCESS_KEY_ID), source.getProperty(SECRET_ACCESS_KEY));
-
-        var client = credentialsValidation.apply(source).succeeded()
-                ? clientProvider.s3Client(source.getProperty(REGION), secretToken)
-                : clientProvider.s3Client(source.getProperty(REGION));
+        S3Client client;
+        var secret = vault.resolveSecret(source.getKeyName());
+        if (secret != null) {
+            var secretToken = typeManager.readValue(secret, AwsTemporarySecretToken.class);
```

Comment:
I think that in the `source` (provider) part, most of the time the credentials won't be "temporary", I would stick with the `AwsSecretToken`, anyway it could be good to handle both.